crates/core/src/operations/update.rs

```diff
@@ -44,76 +112,869 @@ impl Operation for UpdateOp {
+                            "failed broadcasting update change to {} with error {}; dropping connection",
+                            peer.peer,
+                            err
+                        );
+                        // TODO: review this, maybe we should just dropping this subscription
+                        conn_manager.drop_connection(&peer.peer).await?;
+                        incorrect_results += 1;
+                    }
+
+                    broadcasted_to += broadcast_to.len() - incorrect_results;
+                    tracing::debug!(
+                        "Successfully broadcasted update contract {key} to {broadcasted_to} peers - Broadcasting"
+                    );
+
+                    let raw_state = State::from(new_value.clone());
+
+                    let summary = StateSummary::from(raw_state.into_bytes());
+
+                    // Subscriber nodes have been notified of the change, the operation is complete
+                    return_msg = Some(UpdateMsg::SuccessfulUpdate {
+                        id: *id,
+                        target: *upstream,
+                        summary,
+                    });
+
+                    new_state = None;
+                }
+                UpdateMsg::SuccessfulUpdate { id, summary, .. } => {
+                    match self.state {
+                        Some(UpdateState::AwaitingResponse { key, upstream }) => {
+                            tracing::debug!(
+                                tx = %id,
+                                %key,
+                                this_peer = %op_manager.ring.peer_key,
+                                "Peer completed contract value update - SuccessfulUpdate",
+                            );
+
+                            new_state = Some(UpdateState::Finished {
+                                key,
+                                summary: summary.clone(),
+                            });
+                            if let Some(upstream) = upstream {
+                                return_msg = Some(UpdateMsg::SuccessfulUpdate {
+                                    id: *id,
+                                    target: upstream,
+                                    summary: summary.clone(),
+                                });
+                            } else {
+                                // this means op finalized
+                                return_msg = None;
+                            }
+                        }
+                        _ => {
+                            tracing::error!(
+                                state = ?self.state,
+                                "invalid transition in UpdateMsg::SuccessfulUpdate -> match self.state"
+                            );
+
+                            return Err(OpError::invalid_transition(self.id));
+                        }
+                    };
+                }
+                UpdateMsg::UpdateForward {
```

Comment:
Remove this enum variant if we don't need and commeted code

Reply:
sure